docs/database development process/overview.rst

```diff
@@ -0,0 +1,31 @@
+Overview
```

Comment:
Can you remove the spaces from this directory name?

Maybe `docs/db_dev_process/` would be a better name?

Reply:
Yeah, good idea!